link/platform.go

```diff
@@ -0,0 +1,29 @@
+package link
+
+import "runtime"
+
+func byArch(arch_map map[string]string) string {
+	value, ok := arch_map[runtime.GOARCH]
+	if !ok {
+		panic("Unsupported arch " + runtime.GOARCH)
+	}
+	return value
+}
+
+func SysGetpid() string {
```

Comment:
I don't think this warrants a stand alone / exported function, see my comments about archSyscallName and maybe baking this into Kprobe, etc.